apps/src/publicKeyCryptography/ModuloClock.jsx

```diff
@@ -56,32 +60,35 @@ const ModuloClock = React.createClass({
     const maximumTimePerSegment = Math.min(1000, maximumDuration / 10);
 
     this.targetDividend = dividend;
-    this.startTime = Date.now();
     this.interval = setInterval(this.tick, 33);
     this.onStep = onStep || function () {};
     this.onComplete = onComplete || function () {};
     this.duration = Math.min(maximumDuration, dividend * maximumTimePerSegment);
-    this.setState({currentDividend: 0});
+    this.setState({startTime: Date.now(), currentDividend: 0});
   },
 
   tick() {
-    const elapsedTime = Date.now() - this.startTime;
+
+    const elapsedTime = Date.now() - this.state.startTime;
     if (elapsedTime < this.duration) {
       const currentDividend = Math.floor(easeOutCircular(elapsedTime, 0, this.targetDividend, this.duration));
```

Comment:
I'm not sure I understand how the speed is controlled -- each iteration seems to go faster at the beginning and then gradually slow down. Is this what's being done in `easeOutCircular`? It would be nice to briefly mention this in a comment.

One technical comment, in relation to the request to speed up the last step of the animation, would be to put a cap on the length of time each step of the animation can take, maybe 100-200ms.